lib/Serialization/Deserialization.cpp

```diff
@@ -7680,10 +7690,14 @@ Expected<Type> DESERIALIZE_TYPE(SIL_FUNCTION_TYPE)(
   if (!patternSubsOrErr)
     return patternSubsOrErr.takeError();
 
-  auto lifetimeDependenceInfo = MF.maybeReadLifetimeDependenceInfo(numParams);
+  SmallVector<LifetimeDependenceInfo, 1> lifetimeDependencies;
+
+  while (auto lifetimeDependence = MF.maybeReadLifetimeDependence(numParams)) {
+    lifetimeDependencies.push_back(*lifetimeDependence);
+  }
 
-  if (lifetimeDependenceInfo.has_value()) {
-    extInfo = extInfo.withLifetimeDependenceInfo(*lifetimeDependenceInfo);
+  if (!lifetimeDependencies.empty()) {
+    extInfo = extInfo.withLifetimeDependencies(lifetimeDependencies);
```

Comment:
I think you need a `ctx.AllocateCopy(lifetimeDependencies)` here?

Reply:
Fixed